modules/context/repo.go

```diff
@@ -689,24 +682,15 @@ func RepoAssignment(ctx *Context) context.CancelFunc {
 		}
 	}
 
-	// FIXME: use paganation and async loading
-	branchOpts.ExcludeBranchNames = []string{ctx.Repo.Repository.DefaultBranch}
-	brs, err := git_model.FindBranchNames(ctx, branchOpts)
-	if err != nil {
-		ctx.ServerError("GetBranches", err)
-		return cancel
-	}
-	// always put default branch on the top
-	ctx.Data["Branches"] = append(branchOpts.ExcludeBranchNames, brs...)
 	ctx.Data["BranchesCount"] = branchesTotal
 
 	// If not branch selected, try default one.
-	// If default branch doesn't exist, fall back to some other branch.
+	// If default branch doesn't exist, fall back to default branch from setting.
 	if len(ctx.Repo.BranchName) == 0 {
 		if len(ctx.Repo.Repository.DefaultBranch) > 0 && gitRepo.IsBranchExist(ctx.Repo.Repository.DefaultBranch) {
 			ctx.Repo.BranchName = ctx.Repo.Repository.DefaultBranch
-		} else if len(brs) > 0 {
-			ctx.Repo.BranchName = brs[0]
+		} else {
+			ctx.Repo.BranchName = setting.Repository.DefaultBranch
```

Comment:
Is this change right if default branch doesn't exists?

Reply:
I saw similar assignment is also done in [init.go](https://github.com/go-gitea/gitea/blob/d9763d605d5fef058d88cd8b894f99cb9cc06c1c/modules/repository/init.go#L267). Might need confirmation from @lunny and also for this question as well **https://github.com/go-gitea/gitea/pull/25719#discussion_r1264793053**

Reply:
For here, even if the default branch doesn't exist, it should not be handled here. Related further improvements can be done in another PR.